stl/inc/expected

```diff
@@ -685,13 +700,393 @@ public:
+            "expected<T,E>::or_else(F) requires the return type of F to be a specialization of expected. "
+            "(N4928 [expected.object.monadic]/11)");
+        static_assert(is_same_v<typename _Uty::value_type, _Ty>,
+            "expected<T,E>::or_else(F) requires the value type of the return type of F to be T. "
+            "(N4928 [expected.object.monadic]/11)");
+
+        if (_Has_value) {
+            return _Uty(in_place, _Value);
+        } else {
+            return _STD invoke(_STD forward<_Fn>(_Func), _Unexpected);
+        }
+    }
+
+    template <class _Fn>
+        requires is_move_constructible_v<_Ty>
+    constexpr auto or_else(_Fn&& _Func) && {
+        using _Uty = remove_cvref_t<invoke_result_t<_Fn, _Err>>;
+
+        static_assert(_Is_specialization_v<_Uty, expected>,
+            "expected<T,E>::or_else(F) requires the return type of F to be a specialization of expected. "
+            "(N4928 [expected.object.monadic]/15)");
+        static_assert(is_same_v<typename _Uty::value_type, _Ty>,
+            "expected<T,E>::or_else(F) requires the value type of the return type of F to be T. "
+            "(N4928 [expected.object.monadic]/15)");
+
+        if (_Has_value) {
+            return _Uty(in_place, _STD move(_Value));
+        } else {
+            return _STD invoke(_STD forward<_Fn>(_Func), _STD move(_Unexpected));
+        }
+    }
 
+    template <class _Fn>
+        requires is_move_constructible_v<_Ty>
+    constexpr auto or_else(_Fn&& _Func) const&& {
+        using _Uty = remove_cvref_t<invoke_result_t<_Fn, const _Err>>;
+
+        static_assert(_Is_specialization_v<_Uty, expected>,
+            "expected<T,E>::or_else(F) requires the return type of F to be a specialization of expected. "
+            "(N4928 [expected.object.monadic]/15)");
+        static_assert(is_same_v<typename _Uty::value_type, _Ty>,
+            "expected<T,E>::or_else(F) requires the value type of the return type of F to be T. "
+            "(N4928 [expected.object.monadic]/15)");
+
+        if (_Has_value) {
+            return _Uty(in_place, _STD move(_Value));
+        } else {
+            return _STD invoke(_STD forward<_Fn>(_Func), _STD move(_Unexpected));
+        }
+    }
+
+    template <class _Fn>
+        requires is_copy_constructible_v<_Err>
+    constexpr auto transform(_Fn&& _Func) & {
+        static_assert(invocable<_Fn, _Ty&>, "expected<T,E>::transform(F) requires that F is invocable with T. "
+                                            "(N4928 [expected.object.monadic]/19)");
+        using _Uty = remove_cv_t<invoke_result_t<_Fn, _Ty&>>;
+
+        if constexpr (!is_void_v<_Uty>) {
+            static_assert(_Is_invoke_constructible<_Fn, _Ty&>,
+                "expected<T,E>::transform(F) requires that the return type of F is constructible with the result of "
+                "invoking f. "
+                "(N4928 [expected.object.monadic]/19)");

```

Comment:
These can be combined into a single line (clang-format will respect this, although it won't do so automatically). Occurs below.